### ISSUE TITLE
Precourse

### ISSUE THREAD

**Issue by bradreeder** [foundersandcoders/master-reference#26]
There needs to be discussion as to what exactly the precourse materials should be.

At the moment they are:

* HTML, CSS, markdown.
* JS: DOM, weird parts, basic es6.
* command-line, github, git basics
* chrome dev tools
* installation party

The only feedback from FAC9 #18 was that the materials should be compulsory and achievable (not overwhelming) within the time set so that every one enters the course on even footing.

**Comment by jsms90** [comment]
- Current pre-course description of HTML & CSS (for FAC9) is a reference to Free Code Camp, which is now a pre-requisit. Either remove this bullet point from the pre-course, or suggest more complex resources to reflect the current pre-reqs
- I would remove basic es6 if we only intend this to be introduced in week4, and not used before then.
- JS the weird parts - how much of this? the entire thing?
- Command line - if we make specific resources compulsory, we should update week1's readme to add complexity beyond: "Write a tutorial on how to use the command line to find out which directory you're in, how to print its contents, and how to move to another directory"

**Comment by nelsonic** [comment]
Agree with @jsms90  👍 
This is a _good_ list of the _tech_ it would be _really_ good for people to be _familiar_ with `before` they start F&C. ✅ 
@bradreeder is there an issue where there's a list of _non-technical_ (_but no less important_) skills required to be a great "_technologist_" ...? 🤔

**Comment by bradreeder** [comment]
@nelsonic I don't believe there is but feel free to open one if you have something in mind.

**Comment by jsms90** [comment]
@nelsonic Do you mean a publicised list for applicants to look at?

**Comment by nelsonic** [comment]
@jsms90 insomnia? snap! 😉 
***Yes***, I mean a public list e.g: https://github.com/dwyl/start-here/blob/master/books-to-read.md ❤️  ✅  📚

**Comment by bradreeder** [comment]
Only things I've filtered from S-G-C related to precourse:

* More on async functions in precourse materials
* More on git, possibly git udacity course.

**Comment by jsms90** [comment]
@nelsonic are you suggesting giving them a reading list? Or distilling the messages from a reading list, like this, down into what we would consider F&C "values"?

@bradreeder The git udacity course was fairly unanimous as a pre course requirement for F&C10.

Since they have been chosen now, do you know when the F&C10 applicants will receive their pre course list?

**Comment by nelsonic** [comment]
@jsms90 **`Yes`** I'm suggesting we should have a list e.g: https://twitter.com/iteles/status/651409217174290433 _however_ I think people should pick off one or two of the books that _interest_ them so they have _rounded_ knowledge/perspective rather than just focussing on coding skills ... **`And`** I think we can do a _much_ better job of sharing our knowledge/insights derived from our reading ... 😉

**Comment by jsms90** [comment]
I agree. But perhaps this belongs on a site with much more content than ours currently has. I don't like the idea of accidentally communicating the message (even unintentionally) that any reading material that we suggest is in any way necessary for getting onto the course.

**Comment by bradreeder** [comment]
@jsms90 Precourse is generally assigned to a few people from the current F&C, so we'll probably aim to have something together during the first week back.

**Comment by nelsonic** [comment]
@jsms90 no reading material is "_necessary_" or "_required_"
merely "_recommended_" as being "_helpful_", "_useful_" or "_practical_"
for people who want to "_get good_" at making things with technology ...
![image](https://cloud.githubusercontent.com/assets/194400/21339435/2136d138-c677-11e6-9c72-8c009184a653.png)
***Yes***, maybe there's a _better_ place for this list. 👍  (_thanks!_)

**Comment by bradreeder** [comment]
@jsms90 @njsfield @Cleop 

And so it begins!  🙌  :bowtie: 🙌 

I think we should aim to have together a precourse materials `md` file to share with F&C10 by the end of the 2 week break. 

a) So F&C10 have as much time as possible to work through it and so we can begin thinking about inviting them into the space. See https://github.com/foundersandcoders/business/issues/104
b) As it will likely effect how you plan week 1 when we get back.

### Suggestions

My suggestion would be setting some time aside after Christmas to think it through and continue discussion here. Feel free to arrange an independent meeting or skype call amongst yourselves to finalise if that's easier, but aim to post minutes in this issue.

You can make PRs in this repo against this issue and assign to me or @des-des to review. I've been sticking to DWYL's ![contributing](https://github.com/dwyl/contributing) guideline when making PRs here. 

Let me know if you need any guidance from me or if anything is unclear.

### Inspiration

Feel free to re-think what has been done in the past, but if you need ideas or resources to get started: 

##### Sections we've had in the past:

* Learning outcomes
* Resources list
* Exercises
* Installation list/ tools to setup.

##### Past resources:

If you sign into gitbook and go to the ![Founders & Coders](https://www.gitbook.com/@foundersandcoders) organisation you'll be able to find other gitbooks from F&C7 and below. I haven't listed them as I'm unable to open them at the moment (gitbook makes me cry.)

![F&C9 gitbook](https://foundersandcoders.gitbooks.io/fac9/content/PreCourse/)
![F&C8 gitbook](https://foundersandcoders.gitbooks.io/fac8/content/PreCourse/)
![Playbook](https://github.com/foundersandcoders/playbook/blob/master/before-arrival.md)

**Comment by sofer** [comment]
A question for @jsms90 @Cleop and @njsfield: how would you feel about me opening up https://github.com/FAC10 to the new cohort?

**Comment by jsms90** [comment]
@sofer I suggest we do it on the same day that we send out the pre course material. They should have a chance to rest and only spend the next couple of weeks working if they want to, without any indication from us that they should/could be thinking about it before then.
@cleo @njsfield what do you think?

**Comment by njsfield** [comment]
@jsms90 agreed, I think inviting them into [FAC10](https://github.com/FAC10) may accidentally create a bit of premature pressure.

To add to the points listed above, there's a lot of potential resource pooling here and I was wondering whether it may be a good idea to allow the new cohort early access to the udemy login? This is more aimed at allowing full access to the Weird Parts course, as we were recommended the first few hours to watch during pre-course, but some folks on our cohort watched the whole thing because of well taught it is, and well it demystifies a lot of JavaScript concepts. At the very least I'd like to recommend the first 3 hours as we were previously, and perhaps in the README highlight how the Weird Parts teaching style is akin to FAC's teaching style- teaching a broad range of content by carefully journeying through problems/solutions

**Comment by jsms90** [comment]
@sofer it is worth making a F&C10 lobby gitter channel for them though! I have already been messaged by one of them as they have created a channel with most of them in. Let's do this asap and add the pre course mentors? :+1:

**Comment by sofer** [comment]
@jsms90 done.

**Comment by jsms90** [comment]
@njsfield nice idea!! If we set students pre course which is paid for, and we have an account, it seems only fair to give them access.
But maybe we need to be careful about the security of such logins?
It looks like the unofficial meetup group will be continuing in January (FAC10 and non-FAC10), which I couldn't be happier about. If we are un-phased about these login details getting passed around, that's fine. Otherwise, it needs to be made explicit that these details are not to be shared.

**Comment by iteles** [comment]
@jsms90 I don't think we *should* be un-phased, but we seem to be :persevere:

It would make *much* more sense to me if we make our pre-reqs open source (in keeping with our overall mindset on the "democratisation of education").

**Comment by njsfield** [comment]
@iteles I agree that pre-requisites should be open-source (no financial pressure on potential students), and it would be ideal if all FAC members (including FAC10) need only open source materials in their development. Premium content seems to be shared with current FAC students via *udemy* to aid in quicker/deeper understanding of subjects, **but** that can also be attained from open source material (and part of our development is learning how to absorb open source content well in a short space of time). Perhaps in the pre-course material in the Resources section it would be worth mentioning that all resources listed are open source as we’ll be using open source material throughout the course. 

The first 3 hours of *Understanding The Weird parts* is free to view, and many current/past students agree to recommend them, so we could provide good resources for the remaining learning outcomes - *first class functions,  callbacks, closures, IIFEs, prototypal inheritance*. The *Diving into the source code of jQuery* section isn't essential as it explores an advance application of OO JavaScript, but learning from source code is very beneficial, perhaps we could link students to a simple github project and encourage them to try and understand as much as they can after they've watched the 3 hours?

On a side note, I’ve written some thoughts on the practical project recommendations & documents structure in the gitbook;

### Project Exercise
Currently a 'minimum of 100 hours’ is referenced as essential to allowing students ability to tackle Codewars, and the [recommended materials](https://github.com/foundersandcoders/recommended-materials) also mentions Free Code Camp. The Basic Front End Development projects are within the first 100 hours of the Front End Development track, I think it would be reasonable for students to continue with this track up until the *Intermediate Front End Project* section, and make it a requirement to complete just 1 of those projects (Vanilla CSS and JavaScript) as the only creative project required amongst the pre-req material. Students would then not feel pressured to complete too many projects before the course starts, but still have to focus on putting their skills to work to create something that would cushion the blow of Week 1 (HTML/CSS/Toolkit) and Week 2 (APIS). 

### Pre-Course Structure in Gitbook

For FAC9, the structure of the [Pre-Course](https://foundersandcoders.gitbooks.io/fac9/content/PreCourse/) section in the Gitbook is compressed together in a way that the learning outcomes is in the [Resource List](https://foundersandcoders.gitbooks.io/fac9/content/PreCourse/learning_outcomes.html), and the [Exercises](https://foundersandcoders.gitbooks.io/fac9/content/PreCourse/pre-course_materials.html) Section includes a mix of tutorials and recommended practical projects. The [Install List](https://foundersandcoders.gitbooks.io/fac9/content/PreCourse/installation_list.html) has useful links to resources, but has little very little information on the tools themselves.

As @bradreeder mentioned, in the past four sections have been used, perhaps we should use them in the following structure; 

- **Learning Outcomes** - list bullet pointed outcomes organised in sections (*HTML, CSS, Markdown, Dom Manipulation, Callbacks, Closures, Prototypes, Command Line, Github/Git, Dev Tools*)
- **Resource List** - resource(s) listed for each Learning Outcome section (or maybe merge Learning Outcomes & Resource List together?)
- **Exercises** - Practical project exercise(s), which should utilise Learning Outcomes
- **Installation List** - tools to setup, resources on them and how to set them up (we’ll be organising dates/times for FAC10 to come into the space if they’d like to install/work on pre-course materials together, but having additional installation resources in this section would ensure those not able to come into the space would have all they needed).

This assumes we give FAC10 access to the Gitbook next week when we share Pre-course materials?

**Comment by bradreeder** [comment]
@njsfield My only concern with the pre-course material being open-source is that if applicants get through the application criteria sooner than others they may move onto the pre-course before being accepted and arrive more prepared than others. It's important, for this reason, that the precourse materials be designed so that they can be reasonably completed within the time of being accepted and the start of the course.

Also, the master-reference will now be the gitbook in likelihood #28 so if share them the precourse from here you'll just need to notify them that the rest of the repo is under construction until their start date.

**Comment by njsfield** [comment]
@bradreeder sorry I should of clarified- I meant that pre-course material should be *sourced* from open source materials, I completely agree that pre-course material should not be exposed to applicants. 

Two options- 
1. Make a separate repo for FAC10 to view that contains pre-course material. Then when the master reference is ready/gitbooked share it with FAC10 (and at a later point merge the seperate pre-course repo in with the master reference)
2. Finish the pre-course material in this master reference repo, then share with FAC10 and advise that the repo is under construction

I think option 1 would be sensible, it would partially hide the construction distractions, and merging a separate pre-course repo shouldn't be too difficult

**Comment by bradreeder** [comment]
@njsfield I've thought about it and I think it's fine to PR it here personally, but I'm open if others disagree, and we can always share the precourse materials with each F&C through a separate md file in the gitter. It's unavoidable that students may read this repo and the issues as you're constructing your weeks as long as it is public, but I agree we should probably point them away from it until the start.

If we're aiming to make the curriculum open-source through this resource the problem of people potentially looking ahead will be a problem not unique to precourse materials that we'll have to think about, so I don't think I see an issue with the precourse being public in principle. As others in your cohort have said, we should focus on making them achievable in the time-frame so that everyone has enough time to complete them by arrival.

**Comment by njsfield** [comment]
@bradreeder if me @jsms90 and @Cleop aim to have pre-course material ready before the start of next week (ready on Sunday) we can make a decision then where to merge it. Agree that students reading the repo is inevitable, hope the pre-course work will distract them from that curious trail :stuck_out_tongue:

@all what are your thoughts one making a single practical project a pre-course requirement, and the pre-course structure I mentioned in my comment?

**Comment by njsfield** [comment]
Following a google hangouts chat me and @jsms90 made some notes on proposals for pre-course materials;

The materials will consist of three markdown files-
- **Resources** which will include both learning resources and project description, to give directly to FAC10
- **Learning Outcomes** which will be used as reference materials for others when planning weeks
- **Installation Guide** which will also be given directly to FAC10 (and used at the installation party)

# Resources File

Like [here](https://foundersandcoders.gitbooks.io/fac9/content/PreCourse/pre-course_materials.html), but structured as an ordered guide of learning resources which builds up to a practical project description. Most sections will include just one resource.

### Introduction

This will begin with a congratulatory message similar to-
> Congratulations on being part of the next cohort. Hope you’re well rested after working through the pre-requisite material…

Followed by a statement about working through the following material before completing the practical project at the end.
A closing statement will be about using **no frameworks at FAC**, and how the project will be completed with plain HTML, CSS and JavaScript (**no ES6 for browser support reasons**)

### Command Line

[CLI Crash Course](https://learnpythonthehardway.org/book/appendixa.html) (recommended in the past in the [playbook](https://github.com/foundersandcoders/playbook/blob/master/before-arrival.md
 )) will be given as it has very clear instructions and allows students to practice commands as opposed to following simulation tutorials like [here](https://www.codecademy.com/learn/learn-the-command-line
)

### Github

[Udacity: How to Use Git and Github](https://www.udacity.com/course/how-to-use-git-and-github--ud775) will help deepen understanding of version control of git/github, students can simply watch along (and Week 1 will contain a workshop/practical exercises to put knowledge into practice).

### Markdown

[Markdown Tutorial](http://www.markdowntutorial.com/lesson/1/) is a very short set of lessons to practice using markdown syntax.

### CSS

Students will have practiced using some light CSS in pre-requisite material, and will practice more in the practical project…
This section will contain 4-5 links to short articles/tutorials that will be beneficial when tackling CSS throughout the course. I propose the following points to be covered in some short articles/tutorials;
- **Positioning** (namely: centering a div)
- **Using** media queries
- **Specificity** (which should also cover the cascading nature of CSS)
- **Units** (EM/REM/Viewport/percentages)

These articles should also help reduce grey areas of understanding produced from pre-req walkthrough material.

### JavaScript

1. [Understanding The Weird Parts: First 3.5 hours](https://www.youtube.com/watch?v=Bv_5Zv5c-Ts) will help build a deeper understanding of many of JavaScripts core concepts. Followed by...
2. [What the heck is the event loop anyway](https://www.youtube.com/watch?v=8aGhZQkoFbQ) will help give background understanding of event loops and callbacks that should cushion the blow when callbacks are introduced in the first weeks.

### DOM Manipulation

1. [DOM Manipulation Basics](http://callmenick.com/post/basics-javascript-dom-manipulation) will cover a lot of the basics.
2. [DOM Tutorials: Exercises 1,2,2](https://dom-tutorials.appspot.com/static/index.html) will help put that into practice.

### APIs

[Learn APIS: First 4 chapters](https://zapier.com/learn/apis/) introduces basics of APIS, HTTP, and API keys

### Chrome Dev Tools

[Free Code Camp: Chrome Developer Tools](https://www.freecodecamp.com/map) located in the Video Challenges section covers most of Dev Tools, in about 30 minutes worth of video content with quizzes after each video.

### Practical Project

Complete just one of the [Intermediate Front End Developer Projects](https://www.freecodecamp.com/map) using vanilla JavaScript (without ES6 syntax), CSS and HTML. 
Students will have completed a Basic Front End Project during pre-requisites per requirement of 100 hours of study, so an intermediate project should be a sufficient challenge.
However perhaps add;
> If you’ve already completed any of these projects already, complete one of the remaining projects, or come up with your own project that makes an API call.

Some students will be in this situation, some may have completed all of the intermediate projects in which case it should be up to them to decide what project they’d like to complete so they feel challenged and able to practice what they’ve learnt from the above pre-course material.

# Learning Outcomes File

Similar to [here](https://foundersandcoders.gitbooks.io/fac9/content/PreCourse/pre-course_materials.html) but;
- Remove ‘recognise ES6 features’.
- Replace ‘Understanding what a callback is’ with ‘Understanding what the event loop is'. 
- **APIS** should have an additional point- ‘Make an API call and retrieve data’
- Add section **CSS** with several points about understanding of points covered- *Positioning*, *Using*, *Specificity*, *Units*
- Add section **MARKDOWN** with a point about using basic markdown syntax.


# Installation Guide File

Similar to [here](https://foundersandcoders.gitbooks.io/fac9/content/PreCourse/installation_list.html) with a note added to the ‘Install Chrome’ point about [this](https://github.com/atom/atom/releases/tag/v1.12.9) release section for students who need to build Atom from source (linux users).

**Comment by bradreeder** [comment]
@njsfield Looking great! If any of you start on this a folder for you is forthcoming #51 .

How much time do you think you'll need to work on precourse on Monday? Currently dividing the day up for teams to sit together and plan weeks and it looks something like:

> Morning
> 
> Precourse - Nick (clash), Jen, Cleo
> Node 1 & 2 - Nick (clash), Marko, John, Marina, Will, Steve 
> Redis - Nori, Tom, Peter
> API - Ewelina, Emily, Lucy
> 
> Afternoon
> 
> Toolkit - Nick, Jen (clash), Cleo
> Testing - Ewelina, John, Peter
> Postgres - Emily, Tom, Steve
> Hapi templates - Marko, Will, Marina
> Hapi auth - Nori, Jen (clash), Lucy

There's some overlap as it's not possible to arrange teams cleanly. @jsms90 & @Cleop could pair to write-up precourse in the morning, then @jsms90 would need to pick her preference for afternoon (prob toolkit?), and I can try and find someone to fill in for the hapi auth meeting.

**Comment by emilyb7** [comment]
@njsfield @jsms90 @Cleop I really like the choice of resources and topics 👍🏻

However, I also think that this is quite a lot to get through for anyone who has other commitments. 

I like that in our FAC9 pre-course requirements file we were reminded that it's important to come to FAC9 feeling well-rested:

> FAC is a very intense course, and you should make sure you're __well-rested and ready to work when you get here__. However, we would strongly suggest putting the time in to ensure you have a good understanding of a few basics before we get started.

> Some of the recommended resources here will take you less than a couple of hours to complete, while others can take a lot longer. __We're not expecting you to finish everything__, but the course is designed with the assumption that you'll be familiar with these topics by Week One:

I think we should say something like that again so as not to put anyone under unnecessary pressure. And I'm sure that FAC10 are committed / aware enough to make their own decisions about how much time they put into the pre-course materials.

Also, regarding the projects, I agree that doing at least one frontend project in advance is really important. But what do we / FAC10 stand to gain by making this a requirement - especially if they've already done one? (as many of them probably have)

Do we intend to check whether they've actually done it?

I think by way of an incentive to doing (another) project we could create a place for them to share a piece of work that they're proud of, and we could help them by giving some simple feedback / code review.

**Comment by emilyb7** [comment]
Oh another thing: I remember the Zapier tutorial being great but it doesn't actually show you how to make a client-side API request. And Free Code Camp focuses on using JQuery, as do many of the other resources out there. I remember having a lot of problems with the FCC weather project because FCC suggest using the Open Weather API, which is served up via HTTP and is then causes problems both in CodePen and in GitHub pages. Maybe we should provide some extra materials about actually using APIs?

**Comment by njsfield** [comment]
@bradreeder am happy to spend time in the morning working on Node 2 as pre-course material is a lot closer to completion. There are only a few extra resources required that will likely be commented here by tomorrow evening (CSS etc).

@emilyb7 me and @jsms90 discussed the balance of between *requiring* and *recommending* completing of all content and I agree that outside commitments can be an added pressure if we don’t provide a message similar to ‘We're not expecting you to finish everything’. 

I think that not including that message and simply prioritising learning (non-coding) material over practical work would ensure all students are capable of completing it at a linear pace in the time before the course starts. Me and @jsms90 feel that ensuring students are at a similar level at the start for the course is a high priority when going into **Toolkit Week** which is quite packed content wise. I think a message can be added for students to contact us on gitter if they have any problems

> I agree that doing at least one frontend project in advance is really important. But what do we / FAC10 stand to gain by making this a requirement - especially if they've already done one?

Requiring only 1 project could benefit both students who’ve completed an *Intermediate Project* and students who haven’t by letting them practice what they’ve learnt (or built a deeper understanding of) from the learning material (which is why the **resources** document progresses from learning material to the project itself, though it shouldn't be enforced that students should have to complete the learning material before moving on to the project).

> I think by way of an incentive to doing (another) project we could create a place for them to share a piece of work that they're proud of, and we could help them by giving some simple feedback / code review.

Agreed :) Perhaps when posting the materials on gitter next week encourage students not to share their project work until right before/at the start of the course, so we can code review then (encourage students to not post projects early to not create pressure for others?)

> I remember the Zapier tutorial being great but it doesn't actually show you how to make a client-side API request. 

> Maybe we should provide some extra materials about actually using APIs?

I remember this issue! There may be a more suitable resource for APIs that I’ll look into, it would be worth adding a note to encourage students to reach out to us/FAC10 on gitter if they have problems

**Comment by emilyb7** [comment]
@njsfield maybe we can talk about "expectations" - I think that's something between "requirements" and "recommendations". I think it's difficult to make anything an absolute requirement. It's not like we're going to kick them off the course if they actually don't do it.

Back to the final project, I am in no way disputing that it's important for everyone to have had a  go at this. But if someone's done 5 projects on their own already, doing another one is only going to push them farther ahead of the rest of the group. If anyone's already well ahead with pre-course type stuff, I seriously think they should take a break and prioritise having a rest over further learning.

Anyway, most important thing I wanted to say... let's let them all know from the outset that there will be the opportunity to get feedback on their final project if they want it. We can set a deadline for submitting the project to us for feedback, like say 1 week before they start? The only problem there, I now realise, is that we're all off in the week before FAC10, and that's 28 people who might come asking for feedback... maybe 2 weeks before? or too much pressure?

**Comment by emilyb7** [comment]
Disclosure: I didn't do a lot of our pre-course stuff as I was busy moving my life across countries while also having a high-pressure job. One thing I should probably have done more of is reading Eloquent JavaScript. I just discovered that it contains a chapter about API requests, which could be v. useful! http://eloquentjavascript.net/1st_edition/chapter14.html

**Comment by njsfield** [comment]
@emilyb7 I agree using *expectations* would be a sensible terminology as well, perhaps combined with a message to contact us/FAC10 on Gitter if they have any problems will help alleviate pressure, and we’ll have a better understanding of how much of the pre-course material students have completed before the course starts (which will help refactor it in the future).

I think it should be up to students that have already completed several projects how they play the pre-course project, though there are benefits in encouraging them to apply what they’ve learnt studying pre-course material- perhaps we should remove this message
> If you’ve already completed any of these projects already, complete one of the remaining projects, or come up with your own project that makes an API call.

and let students decide.

> I now realise, is that we're all off in the week before FAC10, and that's 28 people who might come asking for feedback.

This could be possible if we (FAC9) assign ourselves each to 3-4 people to provide feedback for their project (the one they decide to share)- and instead of standard Code Review formalities (raising issues etc, which are covered further into the course), we simply provide them short concise feedback via Gitter, resulting in each student receiving feedback from 3-4 of us. I think if this feedback were executed it shouldn’t be in ‘issue’ format i.e. students shouldn’t feel pressured to resolve issues immediately before the course starts, but have some tips that’ll help them going forward. @all what are others thoughts?

Btw, have updated the Document(s) proposal in the comments above to include your suggestions, and CSS tutorials I’ve recommended.

**Comment by emilyb7** [comment]
@njsfield 

> I think it should be up to students that have already completed several projects how they play the pre-course project

agree!

For the code review, if every person in FAC10 were to submit a project, we would need 14 people from FAC9 to review 4 FAC10 projects each in order to guarantee feedback for every student from at least 2 different people. Can we manage that? I think it would be good, but it's up to all of us to get involved.

I also think it would be good to make real GitHub issues, as opposed to Gitter, as it's good practice. We can just be very nice with the issues and emphasise that there's no need to fix the stuff if they don't **want** to.

Were you intending for the project to be done via GitHub anyway, or do it FCC-style? (i.e. Codepen) Given that everyone has had to make a GitHub pages site as part of the application process, they should all feel comfortable enough with using git...

**Comment by njsfield** [comment]
It makes sense to suggest publishing via Github pages, and suggest students like their Github repos when they come to sharing ('Make sure you include a link to the Github pages version').

Agree issues would be better practice if we encourage students to read issues and don't feel pressured to resolve them, plus if several of FAC9 were commenting issues they could see previous ones to ensure no repetition.

Back when we published our pre-course projects to Github Pages it was served over http not https, so some of us simply put the projects in Codepen as that used https which resolved CORS issues with API requests... however now Github Pages is served over https so is all good.

**Comment by jsms90** [comment]
Thanks for the comments @emilyb7 ! Really useful feedback & food for thought.

I definitely agree that "expectations" is probably the best word. But only because, to me, that does not imply that the pre course material is optional. Like you, Emily, I didn't do a lot of our pre course material,. That was because of:
- time pressure: having a full time job & training my replacement
- it was only "recommended" i.e. optional, so I didn't expect anyone teaching us to assume that I had covered all the material. (But I was scared, because I assumed that I would turn up and feel way behind everyone else)
- there was obviously too much for me to get through e.g. the entire Eloquent Javascript book & I thought the FCC project seemed more daunting than it was, but definitely didn't understand API calls. Knowing that I wouldn't be able to complete everything, I chose things that I could tick off and complete in small chunks.

Although the list seems long, the vast majority of the resources we are suggesting are actually pretty short. I think 1.5 months is reasonable to get through this list.

But **I think the pre course should be about filling in potential holes in their knowledge and making sure they are all on the same page before coming to week 1, not setting them extra coding work**. They have already put a lot of coding work in during the pre-reqs and will be doing a lot more during the course. Giving them more preparatory information is one thing, but I think that setting them more practical work to do is the thing that will mean that they are not well rested for the course. I think we're in danger of essentially extending the course, both in terms of them being required to do actual coding work before the start, and by giving mentors work to do before week 1. I think that the only benefit of suggesting that they write more code and build another project is for them to cement the knowledge that they gain from these articles, tutorials, etc.

With that in mind, I suggest that we make the project itself a "recommended" task, and therefore the only thing that can be deemed optional? @njsfield @Cleop @emilyb7 what do you reckon?

**Comment by emilyb7** [comment]
@jsms90 totally agree that it should be about filling holes in knowledge. So if someone already feels confident with one of the topics, they should not feel obliged to read the specific materials that we recommend in too much detail, and can instead focus on other things.

So we **expect** them to know about the topics listed, but not to do any specific tasks.

Also, I agree that it feels a bit like we're extending the course. 

That said, I think that the practical project is the most valuable thing of all. I definitely learn more from doing stuff than from reading stuff. Also, some practical experience is also important in terms of getting everyone on the same page. The intermediate FCC projects don't take too long to complete if you aim to fulfil only the most basic requirements...

I just don't like the idea of setting hard deadlines before the course has even started.

**Comment by jsms90** [comment]
Interesting. Well, maybe the project no more optional than the rest, then. I agree that you generally learn most buy building something yourself.

I understand what you mean, that it is about them having the knowledge, not jumping through hoops. But I don't agree with:
> if someone already feels confident with one of the topics, they should not feel obliged to read the specific materials that we recommend in too much detail, and can instead focus on other things
If we're not recommending the right material, we should choose something else that is better. If we are expecting them to put in more time than is reasonable, we should set them less. But I don't see the point in giving them a whole load preparatory material if it's optional.

Plus, it's nice for everyone to have a frame of reference. For example, if I'm explaining git concepts, I can refer to the stuff in Udacity's course, but only if I can assume that they have all been through it.

**Comment by jsms90** [comment]
Hard deadlines for submitting a project before they're even on the course, no. But 1.5 months to read/watch the above material, yes. If it's too much to expect from people, we shouldn't set it for any of them. But I think 1.5 months is a reasonable time frame to expect _something_ and is nothing but beneficial for them to provide guidance about what to look at in the meantime. It's a way to keep them focused while they wait to start, but without making them overworked.

**Comment by emilyb7** [comment]
Ok, I'm going to try and make this my last comment as this isn't my issue and I'm probs getting over involved, but...

Let's take Udacity as an example, because you ( @jsms90 ) just mentioned it. It's quite long and quite a time commitment. Not necessarily an unrealistic commitment, but still several hours out of someone's life. I would expect that there are people coming onto FAC10 who already know about Git and GitHub Wasn't there recently a whole meetup group doing a workshop about Git? I'm guessing some of those people will be joining us on FAC10. So why make them go through it again? I take your point about being able to refer to it in the course, but if someone does the pre-course materials within the next 2 weeks, will they really remember the specificities of that a month down the line anyway? Doing things and solving problems is more memorable that watching long videos.

My preferred approach would be to provide more detail within our precourse materials file (like a break down of the Udacity course into the points that it covers) so that students have a clear overview of what they will gain from the course and skip to the points that are most valuable to them.

So in short: we should **expect** students to have a solid understanding of Git {break this down into more detailed points} and **recommend** that, in order to achieve this, they complete the Udacity course.

That's it, not going to get involved with that any more. I trust you guys ;)

On a slightly different topic...

@njsfield I had a look at your initial comment about the materials again, and have noticed that you have now suggested swapping Zapier for Eloquent JavaScript. 

I'm on API week (with @lucymonie and @skibinska ) so just to clarify my thoughts on this...

I think the Zapier tutorial is really good and provides a thorough - although not especially technical - introduction to APIs. I think it's aimed more at non-developers, but that's also ideal for incoming students who haven't tackled APIs (or much web development in general) before. For that reason, I think we should leave it in.

The Eloquent Javascript chapter, on the other hand, is quite detail heavy and launches into things like error handling and callbacks pretty quickly. If you know nothing about APIs to start with, it might be a bit daunting. I read it this morning for the first time, and found it really useful in terms of cementing my existing knowledge / clarifying a few details. But I'm already relatively comfortable with working with APIs, so for me it's a totally different situation.

We handle the practical side of making API requests in week 3 anyway, so from my point of view (as an API-week mentor) the Eloquent Javascript chapter doesn't need to be anything more than just recommended extra reading for extra-keen people.

But if you really want the pre-course project to involve using an API, then I do think that we need an additional resource and Eloquent Javascript is simply the best thing I could find during my very brief Google search this morning, so perhaps we should have both on the reading list?

**Comment by jsms90** [comment]
Honestly, get as involved as you like! Doesn't matter if it's not your week/task, at least not to me. It's all good stuff that you're raising anyway!

Udacity & JS the Weird Parts are the two bits that still definitely take sizeable chunks of time! I think the Udacity course made it into the pre-course by popular consensus from our cohort, but probably as the best resource from the list of 4 that we were given in ours. Even with their self-organised workshop, I definitely don't think that many of them currently have a much stronger understanding of git and github than F&C9 had before we started!

But looking at the learning outcomes from our gitbook:
- Understand the purpose of using Git and GitHub
- Create and use a GitHub repository
- Know basic git commands like git add, git commit and git push

Udacity goes way beyond that! They will all know the second two bullet points because of one of their pre-reqs was to make their own github page. I imagine that the first point is still fuzzy for some people. So maybe just an article or something much more concise than an entire Udacity course?

After all, we can do a proper github workshop with them in week 1, to fully introduce them to all of the concepts that are in the second half of the Udacity course. I'm loathe to invent something new when the material has already been done so well by that course, but it is actually really long and involved for pre-course material, and we must be able to adapt the content to better suit a group environment.

That kind of leads on to your point about the material on APIs, because I think we might be putting more into the pre-course than is necessary. I do think that Zapier is enough for them at this level. I don't think they need a more technical introduction in the pre-course. We might as well leave it to week 3 and not overload them.

Actually, I think that one of the advanced projects on FCC might be a better idea than the intermediate ones? Some of us were attempting them during our pre-course time, so I don't think it's too advanced. That way, we avoid suggesting that they make something with API calls when we don't expect them to cover that until week 3 anyway. We might want to go more specific and suggest the calculator or the tic tac toe game, so that people don't do the pomodoro clock along to then see it again in week 2?

**Comment by emilyb7** [comment]
Maybe a choice between the intermediate and advanced FCC projects so as not to put anyone under too much pressure to learn / understand APIs? Honestly, I did a lot of FCC before FAC and I found the noughts and crosses game a lot harder than any of the intermediate projects.

Maybe we can just ask them to watch the first bits of the Udacity course? Then if they like it, they know where to find the rest.

One more than that could potentially come out is the markdown tutorial. I don't think markdown is particularly hard to get your head around. It's far easier than html. An understanding of what it is and where it's useful + access to a markdown reference resource (can be Googled, but I use [this one](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet) ) should be enough for anyone.

And one more thing: I remember there being feedback in weeks 2 / 3 stop-go-continues that some people wish they had better understood the concept of callbacks before the course. If we take out the API stuff, that will no longer be covered. Anyone think that's a problem?

**Comment by bradreeder** [comment]
@emilyb7 I want to encourage cross-pollination in terms of people reading and feeding into each other's work so don't worry about it. 👍 The issues you're assigned to are your core responsibilities, but no one's prohibited from commenting on any issue or pull-request in this repo.

**Comment by Cleop** [comment]
I agree with @emilyb7 and @jsms90 about not overwhelming people with too much content/ not wanting to create too divide people by ability/ not creating a course before the course!

**Comment by Cleop** [comment]
Personally I wouldn't place much emphasis on doing a project before the course. Before FAC I didn't have tonnes of time to do extra stuff and I felt during the course that it was very evident when people had already done things before and I hadn't. Ie. stopwatches or api calls.

Obviously, it's inevitable that some people may have done these things of their own accord but I think if we're _teaching_ these things on the course, the reason we're teaching them is just that, to _teach them something new_.

I think that reading about concepts before the course is useful when dealing with big concepts that take time to digest. But in terms of application, my feeling is that that's what we do on the course. 

I think this is different however in the case of things like markdown where there won't be time dedicated to teaching it on the course. Therefore it is useful for people to have tried it before they start.

**Comment by jsms90** [comment]
@Cleop I think you and I had very similar experiences with the pre-course.
> What we could do to help this however is give a brief summary of each resource

Love it!

> Personally I wouldn't place much emphasis on doing a project before the course. Before FAC I didn't have tonnes of time to do extra stuff and I felt during the course that it was very evident when people had already done things before and I hadn't. Ie. stopwatches or api calls.

That was my thinking around making the project the only optional portion. As @emilyb7 says, building something is an engaging and memorable way of cementing your coding knowledge. And I know @njsfield was very much in favour of the projects. Maybe it comes down to what we consider the point of giving precourse material...


I would keep the markdown tutorial for exactly this reason:
> I think this is different however in the case of things like markdown where there won't be time dedicated to teaching it on the course. Therefore it is useful for people to have tried it before they start.

The link that Emily pointed to is a good for people to refer to, if they have forgotten syntax. But the tutorial is better for teaching it in the first place imo.

> I also think we should be super clear in our outlining of compulsory / recommended tasks. 

Agreed!

**Comment by jsms90** [comment]
Since there is a reminder in the precourse material not to use frameworks, I would quite like to give a bit of an explanation here. This is what I've come up with. Please @des-des @bradreeder @sofer and anyone else, weigh in:

> Please do not forget that we **don't use frameworks at FAC** – ignore Bootstrap and JQuery, and make sure you work in vanilla HTML, CSS and JavaScript. We want you to build your skills without being reliant on any one framework so that you fully appreciate what problems each framework is solving, and are able to therefore make more effective judgements on which technologies to use in future. Another advantage is that it arms students with the flexibility to switch between technologies, as the popularity of each one can shift. And lastly, F&C values are heavily aligned with open source software. You are more able to contribute to and improve upon existing technologies with the fullest understanding of their contents.

**Comment by bradreeder** [comment]
@jsms90 @njsfield @Cleop Would you like me to keep this open in case anything else comes to you or are you happy for me to close? Precourse materials created #57

Awesome work, thank you for taking the time to get this ready so soon.

**Comment by bradreeder** [comment]
Closed #57. Will be shared with F&C9 shortly!
